python/semantic_kernel/agents/agent.py

```diff
@@ -478,3 +562,449 @@ def as_mcp_server(
+        """
+        _preload_builtin_agents()
+
+        agent_type = data.get("type", "").lower()
+
+        if not agent_type:
+            raise AgentInitializationException("Missing 'type' field in agent definition.")
+
+        if agent_type not in AGENT_TYPE_REGISTRY:
+            raise AgentInitializationException(f"Agent type '{agent_type}' is not supported.")
+
+        agent_cls = AGENT_TYPE_REGISTRY[agent_type]
+
+        if not isinstance(agent_cls, DeclarativeSpecProtocol):
+            raise AgentInitializationException(
+                f"Agent class '{agent_cls.__name__}' does not support declarative spec loading."
+            )
+
+        return await agent_cls.from_dict(
+            data,
+            kernel=kernel,
+            settings=settings,
+            **kwargs,
+        )
+
+    @staticmethod
+    async def create_from_file(
+        file_path: str,
+        *,
+        kernel: Kernel | None = None,
+        plugins: list[KernelPlugin | object] | dict[str, KernelPlugin | object] | None = None,
+        settings: "KernelBaseSettings | None" = None,
+        extras: dict[str, Any] | None = None,
+        encoding: str | None = None,
+        **kwargs,
+    ) -> _TAgent:
+        """Create a single agent instance from a YAML file.
+
+        Args:
+            file_path: Path to the YAML file defining the agent.
+            kernel: The Kernel instance to use for tool resolution and agent initialization.
+            plugins: The plugins to use for the agent.
+            settings: The settings to use for the agent.
+            extras: Additional parameters to resolve placeholders in the YAML.
+            encoding: The encoding of the file (default is 'utf-8').
+            **kwargs: Additional parameters passed to the agent constructor if required.
+
+        Returns:
+            An instance of the requested agent.
+
+        Raises:
+            AgentInitializationException: If the file is unreadable or the agent type is unsupported.
+        """
+        _preload_builtin_agents()
+
+        try:
+            if encoding is None:
+                encoding = "utf-8"
+            with open(file_path, encoding=encoding) as f:
+                yaml_str = f.read()
+        except Exception as e:
+            raise AgentInitializationException(f"Failed to read agent spec file: {e}") from e
+
+        return await AgentRegistry.create_from_yaml(
+            yaml_str,
+            kernel=kernel,
+            plugins=plugins,
+            settings=settings,
+            extras=extras,
+            **kwargs,
+        )
+
+
+# endregion
+
+
+# region DeclarativeSpecMixin
+
+_D = TypeVar("_D", bound="DeclarativeSpecMixin")
+
+
+class DeclarativeSpecMixin(ABC):
+    """Mixin class for declarative agent methods."""
+
+    @classmethod
+    async def from_yaml(
+        cls: type[_D],
+        yaml_str: str,
+        *,
+        kernel: Kernel,
```

Comment:
kernel is optional in `create_from_yaml` and in `from_dict` seems inconsistent...

Reply:
By the time we call the (overriden `_from_dict`), we've already handled the plugins by adding them to the kernel, and validated that the spec's tools are present in the kernel, so we don't need to pass the plugins further along.